aptos-move/move-examples/fungible_asset/stablecoin/sources/fiat_token.move

```diff
@@ -0,0 +1,420 @@
+module circle::fiat_token {
+    use aptos_framework::fungible_asset::{Self, MintRef, TransferRef, BurnRef, Metadata};
+    use aptos_framework::object::{Self, Object};
+    use aptos_framework::primary_fungible_store;
+    use std::error;
+    use std::string::{Self, String};
+    use std::option;
+    use std::signer;
+    use std::vector;
+    use aptos_std::smart_table::{Self, SmartTable};
+
+    friend circle::master_minter;
+
+    /// Error codes documentation: https://aptos.dev/reference/error-codes/
+    const EALREADY_INITIALIZED: u64 = 1;
+    const EBLACKLISTED_ADDRESS: u64 = 2;
+    const ECALLER_NOT_MINTER: u64 = 3;
+    const EMINTER_ALLOWANCE_EXCEEDED: u64 = 4;
+    const EINSUFFICIENT_BALANCE: u64 = 5;
+    const ENOT_MASTER_MINTER: u64 = 6;
+    const ENOT_OWNER: u64 = 7;
+
+    /// Asset Symbol
+    const ASSET_SYMBOL: vector<u8> = b"USDC";
+
+    /// Module Config for Initializer
+    struct ModuleConfig has key {
+        is_initialized: bool,
+    }
+
+    /// Store refs to Manage Fiat Token
+    struct FiatTokenConfig has key {
+        mint_ref: MintRef,
+        burn_ref: BurnRef,
+        transfer_ref: TransferRef,
+    }
+
+    /// Store access info to Manage Fiat Token
+    struct FiatTokenAccessInfo has key {
+        minters: SmartTable<address, u64>,
+        blacklist: vector<address>,
+    }
+
+
+    /// TODO: store ALL roles under a named object. 
+
+    /// Store under owner address
+    struct MasterMinterConfig has key {
+        master_minter: address,
+    }
+
+    // struct Blacklister has key, store { }
+    // struct Roles have key { } 
+
+    /// Initialize metadata object and store it in the `Metadata` resource
+    /// This function is called only once when the module is deployed
+    /// Only accessible by the REST API
+    /// Note: owner needs to be the circle account's signer reference
+    entry fun initialize(
+        owner: &signer,
+        master_minter: address,
+        maximum_supply: u128,
+        name: String,
+        symbol: String,
+        decimals: u8,
+    ) {
+        // check if module is already initialized
+        assert!(!exists<FiatTokenConfig>(@circle), error::invalid_state(EALREADY_INITIALIZED));
+        // assert!(!borrow_global<ModuleConfig>(@circle).is_initialized, error::invalid_state(EALREADY_INITIALIZED));
+
+        // move ModuleConfig to owner's account and set `is_initialized` to true
+        move_to(owner, ModuleConfig { is_initialized: true });
```

Comment:
These two perform the same purpose right?

Should it just have the FiatTokenConfig alone?